pkg/controller/nodelifecycle/scheduler/taint_manager_test.go

```diff
@@ -109,60 +111,62 @@ func createNoExecuteTaint(index int) v1.Taint {
 }
 
 // Waits for delete actions to appear and then returns them
-func getDeleteActions(fakeClientset *fake.Clientset) []clienttesting.Action {
-	var actions []clienttesting.Action
-main:
-	for ; ; actions = fakeClientset.Actions() {
-		for _, action := range actions {
-			if action.GetVerb() == "delete" {
-				break main
+func waitForDeleteActionsForPods(fakeClientset *fake.Clientset) bool {
+	err := wait.PollImmediate(10*time.Millisecond, 5*time.Second, func() (bool, error) {
+		for actions := fakeClientset.Actions(); ; actions = fakeClientset.Actions() {
```

Comment:
Why two loops, just `for _, action := range client.Actions() {`

Reply:
We check actions for update